front/pages/api/w/[wId]/data_sources/managed.ts

```diff
@@ -0,0 +1,212 @@
+
+      const providers = await Provider.findAll({
+        where: {
+          workspaceId: owner.id,
+        },
+      });
+      let credentials = credentialsFromProviders(providers);
+
+      const dustDataSource = await DustAPI.createDataSource(
+        dustProject.value.project.project_id.toString(),
+        {
+          dataSourceId: dataSourceName,
+          config: {
+            provider_id: dataSourceProviderId,
+            model_id: dataSourceModelId,
+            splitter_id: "base_v0",
+            max_chunk_size: dataSourceMaxChunkSize,
+            use_cache: false,
+          },
+          credentials,
+        }
+      );
+
+      if (dustDataSource.isErr()) {
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "data_source_error",
+            message: `Could not create the data source. reason: ${dustDataSource.error}`,
+          },
+        });
+      }
+
+      const dataSource = await DataSource.create({
+        name: dataSourceName,
+        description: dataSourceDescription,
+        //assuming managed data sources are always private for now
+        visibility: "private",
+        config: JSON.stringify(dustDataSource.value.data_source.config),
+        dustAPIProjectId: dustProject.value.project.project_id.toString(),
+        workspaceId: owner.id,
+      });
+      const systemAPIKeyRes = await getSystemApiKey(owner.id);
+      if (systemAPIKeyRes.isErr()) {
+        console.error(
+          `Could not create the system API key: ${systemAPIKeyRes.error}`
+        );
+        return apiError(req, res, {
+          status_code: 500,
+          api_error: {
+            type: "internal_server_error",
+            message: "Could not create the system API key",
+          },
+        });
+      }
+
+      try {
+        const connectorsRes = await fetch(
+          `${CONNECTORS_API_URL}/connectors/create/slack`,
+          {
+            method: "POST",
+            headers: {
+              "Content-Type": "application/json",
+            },
+            body: JSON.stringify({
+              workspaceId: owner.id.toString(),
+              APIKey: systemAPIKeyRes.value.secret,
+              dataSourceName: dataSource.name,
+              nangoConnectionId: req.body.nangoConnectionId,
+            }),
+          }
+        );
+        if (!connectorsRes.ok) {
+          return apiError(req, res, {
+            status_code: 500,
+            api_error: {
+              type: "internal_server_error",
+              message: "Failed to create the connector.",
+            },
+          });
+        }
+        const connectorsResJson = (await connectorsRes.json()) as {
+          connectorId: string;
+        };
+
+        dataSource.connectorId = connectorsResJson.connectorId;
+        dataSource.connectorProvider = "slack";
+        dataSource.save();
+
+        res.redirect(`/${owner.sId}/ds/${dataSource.name}`);
+        return;
+      } catch (err) {
```

Comment:
No need for such a try catch if we have good error handling. Not sure what it covers right now?

Reply:
It's a left over from the trpc code. I will remove it 👍